mm2src/coins/nft/nft_structs.rs

```diff
@@ -162,6 +173,24 @@ impl FromStr for Chain {
     }
 }
 
+impl Chain {
+    pub fn from_ticker(s: &str) -> MmResult<Chain, ParseChainTypeError> {
+        match s {
+            "AVAX" => Ok(Chain::Avalanche),
+            "avax" => Ok(Chain::Avalanche),
+            "BNB" => Ok(Chain::Bsc),
+            "bnb" => Ok(Chain::Bsc),
+            "ETH" => Ok(Chain::Eth),
+            "eth" => Ok(Chain::Eth),
+            "FTM" => Ok(Chain::Fantom),
+            "ftm" => Ok(Chain::Fantom),
+            "MATIC" => Ok(Chain::Polygon),
+            "matic" => Ok(Chain::Polygon),
+            _ => MmError::err(ParseChainTypeError::UnsupportedCoinType),
+        }
+    }
```

Comment:
```rust
pub fn from_ticker(s: &str) -> MmResult<Chain, ParseChainTypeError> {
    match s {
        "AVAX" | "avax" => Ok(Chain::Avalanche),
        "BNB" | "bnb" => Ok(Chain::Bsc),
        "ETH" | "eth" => Ok(Chain::Eth),
        "FTM" | "ftm" => Ok(Chain::Fantom),
        "MATIC" | "matic" => Ok(Chain::Polygon),
        _ => Err(MmError::from(ParseChainTypeError::UnsupportedCoinType)),
    }
}
```

Reply:
also, isn't this redundancy? `from_ticker` is basically converting from `&str`
```rust
impl FromStr for Chain {
    type Err = ParseChainTypeError;

    #[inline]
    fn from_str(s: &str) -> Result<Chain, ParseChainTypeError> {
        match s {
            "AVALANCHE" => Ok(Chain::Avalanche),
            "avalanche" => Ok(Chain::Avalanche),
            "BSC" => Ok(Chain::Bsc),
            "bsc" => Ok(Chain::Bsc),
            "ETH" => Ok(Chain::Eth),
            "eth" => Ok(Chain::Eth),
            "FANTOM" => Ok(Chain::Fantom),
            "fantom" => Ok(Chain::Fantom),
            "POLYGON" => Ok(Chain::Polygon),
            "polygon" => Ok(Chain::Polygon),
            _ => Err(ParseChainTypeError::UnsupportedChainType),
        }
    }
}

impl Chain {
    pub fn from_ticker(s: &str) -> MmResult<Chain, ParseChainTypeError> {
        match s {
            "AVAX" => Ok(Chain::Avalanche),
            "avax" => Ok(Chain::Avalanche),
            "BNB" => Ok(Chain::Bsc),
            "bnb" => Ok(Chain::Bsc),
            "ETH" => Ok(Chain::Eth),
            "eth" => Ok(Chain::Eth),
            "FTM" => Ok(Chain::Fantom),
            "ftm" => Ok(Chain::Fantom),
            "MATIC" => Ok(Chain::Polygon),
            "matic" => Ok(Chain::Polygon),
            _ => MmError::err(ParseChainTypeError::UnsupportedCoinType),
        }
    }
}
```

And if there's any reason why you want to keep the two which I can't really think of 😄  you can do this 
```rust
impl Chain {
    pub fn from_ticker(s: &str) -> MmResult<Chain, ParseChainTypeError> { Self::from_str(s).map_to_mm(|err| err) }
}
```

Reply:
> And if there's any reason why you want to keep the two

from str parses `Chain` from blockchain string (I need it for serde),`from_ticker` function parses `Chain` from coin/token name string.
https://github.com/KomodoPlatform/komodo-defi-framework/blob/7df590ffda267b0052462d3b0ca2063b7526bd7a/mm2src/coins_activation/src/eth_with_token_activation.rs#L213

When we parse `Nft` or `NftTransferHistory` structures from Moralis, serde uses from str method. 
Moralis team uses chain params as typical blockchain names in their API. 
`eth` and `bsc` are exceptions. I mean moralis dont use their full names: `ethereum`, `bnb chain` / `binance smart chain`. For BNB Chain they decided to use `bsc`.

Such chain name usage break a rule `"dont confuse coin and blockchain names"` a bit. But I think as eth and bsc/bnb are the standard abbreviations accepted by the community for these blockchains, people usually understand from the context that the topic is about blockchain not coin.
In addition, there are also spaces in `BNB Chain` name, which is inconvenient for use in APIs.

Reply:
I simplified matching in Chain implementations and added doc comments fro from str and from_ticker methods

Reply:
ok nice nice